TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java

```diff
@@ -45,5 +67,35 @@ public void handleScrew() {
             }
         }
     }
+
+    public void handleUBar() {
+        //set U bar to position depending on the uBarNum variable
+        if (uBarNum == 1) {
+            drive(COUNTS_PER_INCH * 20, 1);
+        }
+        if (uBarNum == 2) {
+            drive(COUNTS_PER_INCH * 40, 1);
+        }
+        if (uBarNum == 3) {
+            drive(COUNTS_PER_INCH * 60, 1);
+        }
+        if (uBarNum == 4) {
+            drive(COUNTS_PER_INCH * 80, 1);
+        }
+    }
+
+    private void drive(double uBarTarget, double speed) {
+        uBarLevel += uBarTarget;
```

Comment:
This looks like you're adding the target to the current bar level. But I think you want to just set the target to the new value.

Reply:
I tried removing it and it came up with an error. I don't know why, but the plus sign is definitely necessary.

Reply:
I see the issue.  `uBarLevel` is an integer but `uBarTarget` is a double.  Try making `uBarTarrget` an integer and you should be able to make it `uBarLevel = uBarTarget`